scale-tests/run.sh

```diff
@@ -0,0 +1,434 @@
+readonly DCOS_PASSWORD="${8:-}"
+readonly SECURITY="${9:-permissive}"
+
+for file in "${CLUSTER_SSH_KEY}" "${TEST_CONFIG}"; do
+  if ! [[ -s ${file} ]]; then
+    echo "File '${file}' doesn't exist or is empty"
+    exit 1
+  fi
+done
+
+readonly AWS_ACCOUNT='Team 10'
+readonly CLUSTER_SPARK_PACKAGE_REPO='https://universe-converter.mesosphere.com/transform?url=https://infinity-artifacts.s3.amazonaws.com/permanent/spark/assets/scale-testing/stub-universe-spark.json' # Spark with quota support.
+readonly CONTAINER_NAME="${TEST_NAME}"
+readonly CONTAINER_SSH_AGENT_EXPORTS=/tmp/ssh-agent-exports
+readonly CONTAINER_SSH_KEY=/ssh/key
+readonly IMAGE_NAME="mesosphere/dcos-commons:${TEST_NAME}"
+readonly LOG_FILE="${TEST_NAME}.log"
+readonly TEST_DIRECTORY="${TEST_NAME}"
+
+source "${TEST_CONFIG}"
+
+function log {
+  local -r message="${*:-}"
+  echo "$(date "+%Y-%m-%d %H:%M:%S") | ${message}" 2>&1 | tee -a "${LOG_FILE}"
+}
+
+function container_exec () {
+  local -r command="${*}"
+  local -r full_command="
+    . ${CONTAINER_SSH_AGENT_EXPORTS}
+    ssh-add -k ${CONTAINER_SSH_KEY} > /dev/null 2>&1
+    ${command}
+  "  
+  log "${command}"
+  docker exec "${CONTAINER_NAME}" \
+    bash -c "${full_command}" 2>&1 | tee -a "${LOG_FILE}"
+}
+
+declare -x AWS_PROFILE
+eval "$(maws li "${AWS_ACCOUNT}")"
+
+declare SHOULD_INSTALL_INFRASTRUCTURE
+declare SHOULD_INSTALL_NON_GPU_DISPATCHERS
+declare SHOULD_INSTALL_GPU_DISPATCHERS
+declare SHOULD_RUN_FAILING_STREAMING_JOBS
+declare SHOULD_RUN_FINITE_STREAMING_JOBS
+declare SHOULD_RUN_INFINITE_STREAMING_JOBS
+declare SHOULD_RUN_BATCH_JOBS
+declare SHOULD_RUN_GPU_BATCH_JOBS
+
+echo
+read -p "Install infrastructure? [y/N]: " ANSWER
+case "${ANSWER}" in
+  [Yy]* ) SHOULD_INSTALL_INFRASTRUCTURE=true;;
+  * ) ;;
+esac
+
+echo
+read -p "Install non-GPU dispatchers? [y/N]: " ANSWER
+case "${ANSWER}" in
+  [Yy]* ) SHOULD_INSTALL_NON_GPU_DISPATCHERS=true;;
+  * ) ;;
+esac
+
+echo
+read -p "Install GPU dispatchers? [y/N]: " ANSWER
+case "${ANSWER}" in
+  [Yy]* ) SHOULD_INSTALL_GPU_DISPATCHERS=true;;
+  * ) ;;
+esac
+
+echo
+read -p "Run failing streaming jobs? [y/N]: " ANSWER
+case "${ANSWER}" in
+  [Yy]* ) SHOULD_RUN_FAILING_STREAMING_JOBS=true;;
+  * ) ;;
+esac
+
+echo
+read -p "Run finite streaming jobs? [y/N]: " ANSWER
+case "${ANSWER}" in
+  [Yy]* ) SHOULD_RUN_FINITE_STREAMING_JOBS=true;;
+  * ) ;;
+esac
+
+echo
+read -p "Run infinite streaming jobs? [y/N]: " ANSWER
+case "${ANSWER}" in
+  [Yy]* ) SHOULD_RUN_INFINITE_STREAMING_JOBS=true;;
+  * ) ;;
+esac
+
+echo
+read -p "Run batch jobs? [y/N]: " ANSWER
+case "${ANSWER}" in
+  [Yy]* ) SHOULD_RUN_BATCH_JOBS=true;;
+  * ) ;;
+esac
+
+echo
+read -p "Run GPU batch jobs? [y/N]: " ANSWER
+case "${ANSWER}" in
+  [Yy]* ) SHOULD_RUN_GPU_BATCH_JOBS=true;;
+  * ) ;;
+esac
+
+if docker top "${CONTAINER_NAME}" > /dev/null 2>&1; then
+  log 'Container already running'
+else
+  git clone git@github.com:mesosphere/spark-build.git "${TEST_DIRECTORY}" | tee -a "${LOG_FILE}"
```

Comment:
Why do we need this directory? Why not just use `$(pwd)`?

Reply:
We want to have reproducible test runs, I'll push a thing in a bit: https://github.com/mesosphere/spark-build/pull/348#issuecomment-396608807